### PR TITLE
secrets support

### PR DESCRIPTION
**Pull request by dweomer**
- Add `secrets` references to **Plan** spec. Changes in referenced `secrets` can trigger re-application of a **Plans**.
- The label that a **Plan** places on an applied `node` is no longer the `status.latestVersion` but is instead a sha224 digest of `status.latestVersion` and the key-order-value(s) of each secret's `data` in the order each `secret` is specified. This value is represented as `status.latestHash`

Addresses #6

### CODE REVIEW

File: pkg/upgrade/job/job.go [hunk 1]
```diff
@@ -182,6 +183,9 @@ func NewUpgradeJob(plan *upgradeapiv1.Plan, serviceAccountName, nodeName, contro
 						Env: []corev1.EnvVar{{
 							Name:  "SYSTEM_UPGRADE_PLAN_NAME",
 							Value: plan.Name,
+						}, {
+							Name:  "SYSTEM_UPGRADE_PLAN_LATEST_HASH",
```
Comment (galal-hussein) [inline pkg/upgrade/job/job.go]: Why do you inject the hash as env variable into the the pod?

File: pkg/upgrade/plan/plan.go [hunk 2]
```diff
@@ -94,6 +97,24 @@ func RegisterHandlers(ctx context.Context, serviceAccountName, controllerNamespa
 		return obj, nil
 	})
 
+	// secret events referred to by a plan (potentially) trigger that plan
+	secrets.OnChange(ctx, controllerName, func(key string, obj *corev1.Secret) (*corev1.Secret, error) {
+		if planList, err := plans.Cache().List(controllerNamespace, labels.Everything()); err != nil {
+			logrus.Error(err)
```
Comment (galal-hussein) [inline pkg/upgrade/plan/plan.go]: shouldn't this return nil, err

File: pkg/upgrade/job/job.go (hunk 1 above)
Comment (dweomer) [reply pkg/upgrade/job/job.go]: It's general purpose and something might need it. 🤷‍♂ 

The latest version is/was there so I decided to replicate that behavior with the hash.

File: pkg/upgrade/plan/plan.go (hunk 2 above)
Comment (dweomer) [reply pkg/upgrade/plan/plan.go]: That is a good question. I did intentionally write this secrets-trigger-plan logic to make a best effort attempt and I figured if listing secrets is causing problems it is probably a permissions problem and therefore not something likely to resolve itself (thus the re-enque implied by returning an error seemed gratuitous). I am open to changing this.

Comment (galal-hussein) [reply pkg/upgrade/plan/plan.go]: I am just wondering what will be the result of returning nil object for both the secret and the error to the controller when a listing error happen?

Comment (dweomer) [reply pkg/upgrade/plan/plan.go]: It should have fell through to the last line of the function that was returning `obj, nil`. I ran your comment by @ibuildthecloud and he agreed so I changed both the secret and node OnChange events to return the potential error when listing plans to (potentially) trigger

**Pull request merged**